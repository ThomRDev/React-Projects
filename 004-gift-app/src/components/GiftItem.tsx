import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

interface IGiftItemProps {
    src : string
    title : string
}

const GiftItemEl = styled.div`
  animation: fadeIn 0.5s ease forwards;
  position: relative;
  h2{
    position: absolute;
    bottom:0 ;
    left: 0;
    width: 100%;
    text-align: center;
    color: white;
    margin: 0;
    padding: 1em;
    font-family: monospace;
    background:linear-gradient(rgba(255,255,255,0),rgba(0,0,0,0.6) 70%,black);
  }
`

GiftItemEl.displayName="div"

const GiftItem = ({ src,title }:IGiftItemProps) => {

  const hanldeLoad = () => {

  }

  return (
    <GiftItemEl className='item'>
      <img src={src} alt={title} onLoad={ hanldeLoad } style={{ width:"100%",height:"100%",objectFit:"cover" }}/>
      <h2>{title}</h2>
    </GiftItemEl>
  )
}

GiftItem.propTypes = {
    src : PropTypes.string.isRequired,
    title : PropTypes.string.isRequired
}

export default GiftItem