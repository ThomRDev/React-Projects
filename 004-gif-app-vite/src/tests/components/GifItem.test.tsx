import { render,screen } from "@testing-library/react"
import GiftItem from "../../components/GifItem"
describe('Test in <GifItem /> component', () => {

  const props = {
    title : "The Beatles",
    src : "https://media4.giphy.com/media/3oz8xVPdsiFn9F6s4o/giphy-downsized-medium.gif?cid=ce089fd5s5378q9iobbz6jjrj8bl5y5f4fqdxpcx8152o8oq&rid=giphy-downsized-medium.gif&ct=g"
  }

  test('Should evalute the snapshot', () => {
    const { container } = render(<GiftItem { ...props } />)
    expect(container).toMatchSnapshot()
  })
  test('the title of the item should be '+props.title, () => { 
    
    render(<GiftItem { ...props } />)
    expect(screen.getByText(props.title)).toBeTruthy()
    expect(screen.getByText(props.title).innerHTML).toBe(props.title)
  })
  
  test('should show the imagen and the alt attribute', () => {
    render(<GiftItem { ...props } />)
    const imgElement = (screen.getByRole("img") as HTMLImageElement)
    expect(imgElement.src).toBe(props.src)
    expect(imgElement.alt).toBe(props.title)
  })
})