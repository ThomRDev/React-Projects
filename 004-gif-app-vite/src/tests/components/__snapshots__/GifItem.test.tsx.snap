// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem /> component Should evalute the snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ dRAxHT item"
  >
    <img
      alt="The Beatles"
      src="https://media4.giphy.com/media/3oz8xVPdsiFn9F6s4o/giphy-downsized-medium.gif?cid=ce089fd5s5378q9iobbz6jjrj8bl5y5f4fqdxpcx8152o8oq&rid=giphy-downsized-medium.gif&ct=g"
      style="width: 100%; height: 100%; object-fit: cover;"
    />
    <h2>
      The Beatles
    </h2>
  </div>
</div>
`;
